### PR TITLE
Source Instatus: built in yaml

### LINKED ISSUE #14046: New source: connector InStatus
## Tell us about the problem you're trying to solve
If we need to keep track of uptime from various components.
These metrics can be derived from the status page.

## Describe the solution you’d like
Retrieve those numbers from the InStatus API back to the data warehouse to feed dashboards.

## Describe the alternative you’ve considered or used
Manually update OKR numbers.

### PR DESCRIPTION
**Pull request by darynaishchenko**
Added **source-instatus**

**Added two custom components ListAddFields and UpdatesSubstreamSlicer.** 
This was done for the _maintenance_updates_stream_ and _incident_updates_stream_ to work correctly. 
The work algorithm is as follows, in the parent streams maintenance and incident is a list of all update objects. The ListAddFields component selects their IDs from these objects and adds them to the new list field "updates_ids". 
Next, UpdatesSubstreamSlicer creates a stream of slices, iterating over the list of updates_ids and adding them to the slice with the correct parent id.
That is, ListAddFields solves the problem of not having a separate list of all updates_ids at the beginning and cannot extract them from the list of update objects. And UpdatesSubstreamSlicer has an additional iteration over the list of updates and creates a correct stream slice.

### CODE REVIEW

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/incidents.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/incidents.json:66]: nit: please add new line

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/maintenance_updates.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/maintenance_updates.json:56]: nit: please add new line

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/maintenances.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/maintenances.json:79]: Same.

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/metrics.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/metrics.json:36]: Same.

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/page_components.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/page_components.json:87]: Same.

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/public_data.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/public_data.json:40]: new line

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/subscribers.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/subscribers.json:47]: new line

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/templates.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/templates.json:80]: new line

File: airbyte-integrations/connectors/source-instatus/source_instatus/schemas/user.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/schemas/user.json:21]: new line

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml:38]: Does API return the `RetryAfter` header for 429?

**Review by bazarnov — CHANGES_REQUESTED** [review]
Changes are required. Please check the comments bellow.

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: The documentation doesn't have detailed information about the errors, so all I could do was handle the errors I encountered. 401 replaces 403 and 422 errors.

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: Yes, thank you, I replaced `ConstantBackoffStrategy` with `WaitTimeFromHeader`.

**Review by bazarnov — APPROVED** [review]
LGTM, once custom components are covered with unit/integration tests. Nice!

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (brianjlai) [inline airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml:19]: It looks like the `public_data_requester` is duplicating almost everything from the `requester`/`api_error_handler_requester`. The main thing that appears to change is that the url_base is set to `https://{{ stream_slice.url }}.instatus.com/summary.json` instead. Can we instead refactor this a bit so that everything can be defined on `requester` and reduce the duplicated fields.. 

The `public_data_requester` can just be written to override the one `url_base` instead of the whole thing:

```
public_data_requester:
  $ref: "*ref(definitions.requester)"
  url_base: "https://{{ stream_slice.url }}.instatus.com/summary.json"
```

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (brianjlai) [inline airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml:171]: nit extra space

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (brianjlai) [inline airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml:269]: nit extra space

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (brianjlai) [inline airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml:265]: To stick with convention we should separate the `base_url` and the `path`:
`url_base: "https://{{ stream_slice.url }}.instatus.com/"`
`path: "summary.json"`

File: airbyte-integrations/connectors/source-instatus/source_instatus/components.py
Comment (brianjlai) [inline airbyte-integrations/connectors/source-instatus/source_instatus/components.py:18]: For both of these custom components can you please include a quick summary in the PR description and in these comments why the existing `AddFields` component and the `UpdatesSubstreamSlicer` were not suitable for reuse by the Instatus API?

**Review by brianjlai — COMMENTED** [review]
Overall looks good but I added a couple of suggestions on how we might streamline some of the components which had some repitition. Along with some formatting nits.

Also echoing @bazarnov to please add unit tests for the custom components and explain in the PR description why they were needed over the existing ones in the language. This feedback helps us figure out how we can enhance the language. Once we have those I will approve

File: airbyte-integrations/connectors/source-instatus/source_instatus/components.py
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/components.py:18]: Added two custom components ListAddFields and UpdatesSubstreamSlicer. 
This was done for the maintenance_updates_stream and incident_updates_stream to work correctly. 
The work algorithm is as follows, in the parent streams maintenance and incident is a list of all update objects. The ListAddFields component selects their IDs from these objects and adds them to the new field "updates_ids", which is a list. 
Next, UpdatesSubstreamSlicer creates a stream of slices, additionally iterating over the list of updates_ids and adding them to the slice with the correct parent id.
That is, ListAddFields solves the problem that we do not have a separate list of all updates_ids at the beginning and cannot extract them from the list of update objects. And UpdatesSubstreamSlicer has an additional iteration over the list of updates and creates a correct stream slice.

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: done

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: done

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: done

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: done

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: done

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: done

File: airbyte-integrations/connectors/source-instatus/setup.py
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/setup.py]: done

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (brianjlai) [inline airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml:83]: instead of defining this in every stream, if you were to reference the schema loader in the base stream, you should be able to avoid defining it in every stream.

```
base_stream:
  retriever:
    $ref: "*ref(definitions.retriever)"
  schema_loader:
    $ref: "*ref(definitions.schema_loader)"
```

**Review by brianjlai — COMMENTED** [review]
Thank you for adding comments for the custom components and for updating the connector manifest. One more suggestion to reuse the `schema_loader` component, and once the `TestBasicRead` and `TestFullRefresh ` are passing then this is ready to be released.

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml]: thanks, done

File: airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml:239]: I returned the previous version because it doesn't have `stream_slice` at the url_base step, so I received an error while SAT.

Comment (brianjlai) [reply airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml:239]: ah i see. sorry for the mistake

**Review by brianjlai — APPROVED** [review]
Changes look good!

**Pull request merged**
